diff --git a/src/libcamera/pipeline/rpi/vc4/vc4.cpp b/src/libcamera/pipeline/rpi/vc4/vc4.cpp
index c6f39023..f69f2dd4 100644
--- a/src/libcamera/pipeline/rpi/vc4/vc4.cpp
+++ b/src/libcamera/pipeline/rpi/vc4/vc4.cpp
@@ -411,8 +411,11 @@ int PipelineHandlerVc4::platformRegister(std::unique_ptr<RPi::CameraData> &camer
 
 	for (auto stream : data->streams_) {
 		int ret = stream->dev()->open();
-		if (ret)
+		if (ret) {
+			LOG(RPI, Error) << "Failed to open stream "
+					<< stream->dev()->deviceNode() << ": " << ret;
 			return ret;
+		}
 	}
 
 	if (!data->unicam_[Unicam::Image].dev()->caps().hasMediaController()) {
diff --git a/src/libcamera/pipeline/rpi/common/pipeline_base.cpp b/src/libcamera/pipeline/rpi/common/pipeline_base.cpp
index 5d726937..a917bd89 100644
--- a/src/libcamera/pipeline/rpi/common/pipeline_base.cpp
+++ b/src/libcamera/pipeline/rpi/common/pipeline_base.cpp
@@ -822,8 +822,13 @@ int PipelineHandlerBase::registerCamera(std::unique_ptr<RPi::CameraData> &camera
 	 * chain. There may be a cascade of devices in this chain!
 	 */
 	MediaLink *link = sensorEntity->getPadByIndex(0)->links()[0];
-	if (!data->enumerateVideoDevices(link, frontendName))
+	if (!data->enumerateVideoDevices(link, frontendName)) {
+		LOG(RPI, Error) << "Cannot enumerate camera path from "
+				<< link->source()->entity()->name() << " to "
+				<< link->sink()->entity()->name() << " (function "
+				<< link->sink()->entity()->function() << ")";
 		return -EINVAL;
+	}
 
 	ipa::RPi::InitResult result;
 	if (data->loadIPA(&result)) {
