diff --git a/src/libcamera/pipeline/rpi/common/pipeline_base.cpp b/src/libcamera/pipeline/rpi/common/pipeline_base.cpp
--- a/src/libcamera/pipeline/rpi/common/pipeline_base.cpp
+++ b/src/libcamera/pipeline/rpi/common/pipeline_base.cpp
@@ -1074,34 +1074,23 @@
 	    entity->function() != MEDIA_ENT_F_VID_IF_BRIDGE)
 		return false;
 
-	/* Find the source pad for this Video Mux or Bridge device. */
-	const MediaPad *sourcePad = nullptr;
-	for (const MediaPad *pad : entity->pads()) {
-		if (pad->flags() & MEDIA_PAD_FL_SOURCE) {
-			/*
-			 * We can only deal with devices that have a single source
-			 * pad. If this device has multiple source pads, ignore it
-			 * and this branch in the cascade.
-			 */
-			if (sourcePad)
-				return false;
-
-			sourcePad = pad;
-		}
-	}
-
+	/* Follow each source pad until the requested frontend is found. */
 	LOG(RPI, Debug) << "Found video mux device " << entity->name()
-			<< " linked to sink pad " << sinkPad->index();
+				<< " linked to sink pad " << sinkPad->index();
 
 	bridgeDevices_.emplace_back(std::make_unique<V4L2Subdevice>(entity), link);
 	bridgeDevices_.back().first->open();
 
-	/*
-	 * Iterate through all the sink pad links down the cascade to find any
-	 * other Video Mux and Bridge devices.
-	 */
-	for (MediaLink *l : sourcePad->links()) {
-		frontendFound = enumerateVideoDevices(l, frontend);
+	for (const MediaPad *pad : entity->pads()) {
+		if (!(pad->flags() & MEDIA_PAD_FL_SOURCE))
+			continue;
+
+		for (MediaLink *l : pad->links()) {
+			frontendFound = enumerateVideoDevices(l, frontend);
+			if (frontendFound)
+				break;
+		}
+
 		if (frontendFound)
 			break;
 	}
